packages/client/hooks/useRecordIdsWithStages.ts

```diff
@@ -1,25 +1,31 @@
+import graphql from 'babel-plugin-relay/macro'
 import {useMemo} from 'react'
-import {NewMeetingPhaseTypeEnum} from '../__generated__/ParabolScopingSearchResults_meeting.graphql'
-import {ParabolScopingSearchResults_meeting} from '../__generated__/ParabolScopingSearchResults_meeting.graphql'
-import {JiraScopingSearchResults_meeting} from '../__generated__/JiraScopingSearchResults_meeting.graphql'
+import {readInlineData} from 'react-relay'
+import {useRecordIdsWithStages_phase} from '../__generated__/useRecordIdsWithStages_phase.graphql'
 
-const useRecordIdsWithStages = (
-  phases:
-    | ParabolScopingSearchResults_meeting['phases']
-    | JiraScopingSearchResults_meeting['phases'],
-  phaseType: NewMeetingPhaseTypeEnum,
-  recordName: 'issue' | 'task'
-): Set<string> => {
-  const targetPhase = (phases as ReadonlyArray<any>).find((phase) => phase.phaseType === phaseType)!
-  const {stages} = targetPhase
+const useRecordIdsWithStages = (phaseRef: any) => {
   return useMemo(() => {
+    const estimatePhase = readInlineData<useRecordIdsWithStages_phase>(
```

Comment:
hey @tiffanyhan here's a little-known pattern that makes this function a little more generic. Relay has this `readInlineData` function where you can declare the fields you want inline. This is great for functions that don't live in the component file because it uncouples them & the data requirements are co-located with the logic. No right or wrong, just another something for the tool belt!